datalake/src/main/java/com/sequenceiq/datalake/service/sdx/SdxService.java

```diff
@@ -676,29 +677,29 @@ private void validateEnv(DetailedEnvironmentResponse environment) {
     }
 
     private StackV4Request getStackRequest(SdxClusterShape shape, boolean razEnabled, StackV4Request internalStackV4Request, CloudPlatform cloudPlatform,
-            String runtimeVersion, ImageSettingsV4Request imageSettingsV4Request) {
+            String runtimeVersion, ImageSettingsV4Request imageSettingsV4Request, Integer javaVersion) {
+        StackV4Request stackV4Request = internalStackV4Request;
         if (internalStackV4Request == null) {
-            StackV4Request stackRequest = cdpConfigService.getConfigForKey(
+            stackV4Request = cdpConfigService.getConfigForKey(
                     new CDPConfigKey(cloudPlatform, shape, runtimeVersion));
-            if (stackRequest == null) {
+            if (stackV4Request == null) {
                 LOGGER.error("Can't find template for cloudplatform: {}, shape {}, cdp version: {}", cloudPlatform, shape, runtimeVersion);
                 throw new BadRequestException("Can't find template for cloudplatform: " + cloudPlatform + ", shape: " + shape +
                         ", runtime version: " + runtimeVersion);
             }
-            stackRequest.getCluster().setRangerRazEnabled(razEnabled);
-
             if (imageSettingsV4Request != null) {
-                stackRequest.setImage(imageSettingsV4Request);
+                stackV4Request.setImage(imageSettingsV4Request);
             }
-
-            return stackRequest;
-        } else {
-            // We have provided a --ranger-raz-enabled flag in the CLI, but it will
-            // get overwritten if you use a custom json (using --cli-json). To avoid
-            // this, we will set the raz enablement here as well. See CB-7474 for more details
-            internalStackV4Request.getCluster().setRangerRazEnabled(razEnabled);
-            return internalStackV4Request;
         }
+
+        // We have provided a --ranger-raz-enabled flag in the CLI, but it will
+        // get overwritten if you use a custom json (using --cli-json). To avoid
+        // this, we will set the raz enablement here. See CB-7474 for more details
+        stackV4Request.getCluster().setRangerRazEnabled(razEnabled);
```

Comment:
I think this comment is no longer relevant as it was moved out of the if condition

Reply:
It describes why we need to set the flag on top op the provided stackv2request. I think it it still relevant.